code/modules/reagents/chemistry_machinery/chem_dispenser.dm

```diff
@@ -214,8 +216,8 @@
 		if(accept_beaker_only && istype(attacking_object,/obj/item/reagent_container/food))
 			to_chat(user, SPAN_NOTICE("This machine only accepts beakers"))
 			return
-		if(pressurized_only && !istype(attacking_object, /obj/item/reagent_container/glass/pressurized_canister))
-			to_chat(user, SPAN_NOTICE("This machine only accepts pressurized canisters"))
+		if(pressurized_only && !(attacking_object.type in whitelisted_containers))
```

Comment:
if(length(whitelisted_containers) && !(attacking_object.type in whitelisted_containers))
this way the check only succeeds if there is something in the whitelist and is otherwise skipped